frontend/packages/client/src/api/notificationService.js

```diff
@@ -0,0 +1,117 @@
+import {
+  LP_APP_ID,
+  LP_CRO_KEY,
+  LP_EXPORT_KEY,
+} from 'api/constants';
+import Leanplum from 'leanplum-sdk';
+
+const COMMUNITY_UPDATES_CATEGORY_ID = 1;
+const options = {
+  method: 'GET',
+  headers: { accept: 'application/json' },
+};
+
+export const setUserId = async (walletId) => {
+  try {
+    Leanplum.setUserId(walletId);
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const getUserSettings = async (walletId) => {
+  try {
+    fetch(
+      `https://api.leanplum.com/api?appId=${LP_APP_ID}&clientKey=${LP_EXPORT_KEY}&apiVersion=1.0.6&userId=${walletId}&action=exportUser`,
+      options
+    ).then((response) => {
+      let data = response.response[0];
+      let communitySubscriptions = [];
+      let isSubscribedToCommunityUpdates = true;
+      const res = {
+        email: data.userAttributes.email,
+      };
+
+      for (const property in data.userAttributes) {
+        if (property.includes('community')) {
+          let communityId = property.split('community')[1];
+          communitySubscriptions.push({
+            communityId,
+            subscribed: data.userAttributes[property] === 'True',
+          });
+        }
+      }
+
+      res.communitySubscription = communitySubscriptions;
+
+      if (data.unsubscribeCategories) {
+        data.unsubscribeCategories.forEach((category) => {
+          if (category.id == 1) {
+            isSubscribedToCommunityUpdates = false;
+          }
+        });
+      }
+
+      res.isSubscribedToCommunityUpdates = isSubscribedToCommunityUpdates;
+      return res;
+    });
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const setUserEmail = async (email) => {
+  try {
+    Leanplum.setUserAttributes({ email });
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const unsubscribeCommunity = async (communityId) => {
+  try {
+    let userAttributes = {};
+    let communityKey = 'community' + communityId;
+    userAttributes[communityKey] = false;
+    Leanplum.setUserAttributes(userAttributes);
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const subscribeCommunity = async (communityId) => {
+  try {
+    let userAttributes = {};
+    let communityKey = 'community' + communityId;
+    userAttributes[communityKey] = true;
+    Leanplum.setUserAttributes(userAttributes);
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const unsubscribeFromEmailNotifications = async (walletId) => {
+  try {
+    fetch(
+      `https://www.leanplum.com/api?appId=${LP_APP_ID}&clientKey=${LP_CRO_KEY}&action=setUserAttributes&userId=${walletId}&unsubscribeCategoriesToAdd=${COMMUNITY_UPDATES_CATEGORY_ID}`,
+      options
+    ).then((response) => {
+      return response;
+    });
+  } catch (e) {
+    throw new Error(e);
+  }
+};
```

Comment:
I think the fetch request might need a tweak: 
The Promise returned from fetch() won't reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, as soon as the server responds with headers, the Promise will resolve normally (with the ok property of the response set to false if the response isn't in the range 200–299), and it will only reject on network failure or if anything prevented the request from completing.


```
try {
  const response = await fetch('....');
  if (!response.ok) {
    // request didn't work
    // throw an error 
  }
return response;
} catch (error) {
  console.log('there is an error  ');
  // do something with the error but do not bubble it up otherwise do not catch it and leave the caller function to handle it
}
```